### PR TITLE
add support for f8 on gfx1200 gfx1201

### PR DESCRIPTION
**Pull request by jeffdaily**

### CODE REVIEW

**Comment by rocm-mici** [comment]
Jenkins build for [fbcfdcaee6ff14edc744890438ee0ebe64afa2d1](https://github.com/ROCm/pytorch/commit/fbcfdcaee6ff14edc744890438ee0ebe64afa2d1) commit finished as FAILURE
Links: [Blue Ocean view](http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/pytorch%2Fpytorch-ci-pipeline/detail/PR-1603/1//pipeline) / [Build artifacts](http://ml-ci-internal.amd.com:8080/job/pytorch/job/pytorch-ci-pipeline/job/PR-1603/1/artifact)

File: aten/src/ATen/Context.cpp [hunk 1]
```diff
@@ -282,9 +282,9 @@ at::BlasBackend Context::blasPreferredBackend() {
 #ifdef USE_ROCM
   if (blas_preferred_backend == at::BlasBackend::Cublaslt) {
     static const bool hipblaslt_unsupported = []() {
-      static const std::vector<std::string> archs = {"gfx90a", "gfx940", "gfx941", "gfx942"};
+      static const std::vector<std::string> archs = {"gfx90a", "gfx940", "gfx941", "gfx942", "gfx1200", "gfx1201"};
       for (auto index: c10::irange(getNumGPUs())) {
-        if (!detail::getCUDAHooks().isGPUArch(index, archs)) {
+        if (!detail::getCUDAHooks().isGPUArch(static_cast<c10::DeviceIndex>(index), archs)) {
```
Comment (alugorey) [inline aten/src/ATen/Context.cpp:287]: More for my own enlightenment, why did we need to add the static_cast?

**Review by alugorey — APPROVED** [review]
LGTM! Thanks for doing the heavy lifting on this one @jeffdaily! I owe ya one!

File: aten/src/ATen/Context.cpp (hunk 1 above)
Comment (jeffdaily) [reply aten/src/ATen/Context.cpp:287]: The linter complained about unsigned to signed cast here and the linter made this suggestion.  I blindly followed.

**Comment by rocm-mici** [comment]
Jenkins build for [fbcfdcaee6ff14edc744890438ee0ebe64afa2d1](https://github.com/ROCm/pytorch/commit/fbcfdcaee6ff14edc744890438ee0ebe64afa2d1) commit finished as FAILURE
Links: [Blue Ocean view](http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/pytorch%2Fpytorch-ci-pipeline/detail/PR-1603/2//pipeline) / [Build artifacts](http://ml-ci-internal.amd.com:8080/job/pytorch/job/pytorch-ci-pipeline/job/PR-1603/2/artifact)

**Review by pruthvistony — APPROVED** [review]

**Comment by pruthvistony** [comment]
@alugorey ,
Please cherry-pick this change into the rocm6.3_internal_testing_gfx950 branch on pytorch-private repo.
Then this change will be part of MI350 docker images.

**Pull request merged**